lib/Backend/FlowGraph.cpp

```diff
@@ -1721,21 +1767,26 @@ FlowGraph::Destroy(void)
                         Assert(!this->func->DoGlobOpt());
                         break;
                     case Js::OpCode::BrOnNoException:
-                        Assert(region->GetType() == RegionTypeTry || region->GetType() == RegionTypeCatch || region->GetType() == RegionTypeFinally);
+                        Assert(region->GetType() == RegionTypeTry || region->GetType() == RegionTypeCatch || region->GetType() == RegionTypeFinally ||
+                            (predRegion->GetType() == RegionTypeFinally && predBlock->GetLastInstr()->AsBranchInstr()->m_brFinallyToEarlyExit));
```

Comment:
Added comment and test case for this. This happens when the early exit code is moved inside the finally block. And the early exit has 2 edges one from a root region and one from the preceeding BrOnNoException block.